corehq/apps/users/models.py

```diff
@@ -800,6 +813,8 @@ class CouchUser(Document, DjangoUserMixin, IsMemberOfMixin, UnicodeMixIn, EulaMi
     assigned_location_ids = StringListProperty()
     has_built_app = BooleanProperty(default=False)
 
+    reporting_metadata = SchemaProperty(ReportingMetadata)
```

Comment:
i did this way because i knew it'd make the query easier on the report side since i was not sure how much ES supported doing joins. the reason it's semi-important that this be able to be done in one query is so we can grab just the top 10 users and sort by one of those properties (currently that the reason we have to fetch every user). would be interested to see how much these kinds of extra properties affect the load time. my intuition was that adding things like string/integers are fairly lightweight

there looks to be some support for ES joins, but it's not that fully featured: https://www.elastic.co/guide/en/elasticsearch/reference/current/joining-queries.html